src/braket/circuits/circuit.py

```diff
@@ -196,6 +201,28 @@ def qubits(self) -> QubitSet:
         """QubitSet: Get a copy of the qubits for this circuit."""
         return QubitSet(self._moments.qubits.union(self._qubit_observable_set))
 
+    @property
+    def has_free_parameters(self) -> bool:
+        """
+        This is a check to see if the circuit has any free parameters.
+        This can be used to block executions of parameterized circuits without
+        iterating through the circuit each attempted execution.
+
+        Returns:
+            bool: True if the circuit has any free parameters.
+        """
+        return self._parameterized
+
+    @property
+    def parameters(self) -> Set[FreeParameter]:
+        """
+        Gets a set of the parameters in the Circuit.
+
+        Returns:
+            Set[FreeParameter]: The `FreeParameters` in the Circuit.
+        """
+        return self._parameters
```

Comment:
Is self._parameters == (self._parameters != None) ?

Reply:
Can you explain why we actually need to keep track of this? I believe it's not just for optimization purposes, right?

Reply:
This is for tracking the parameters to update later on. Additionally, customers may try and access circuit free parameters later on. It can be that the set is empty if there are no free parameters in the circuit.